packages/flutter/test/painting/border_side_test.dart

```diff
@@ -0,0 +1,110 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/painting.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  test('BorderSide - asserts when constructed incorrectly', () {
+    expect(
+      const BorderSide(),
+      const BorderSide(
+        color: const Color(0xFF000000),
+        width: 1.0,
+        style: BorderStyle.solid,
+      ),
+    );
+    // so that we can use `new` below, we use these:
+    final Null $null = null;
```

Comment:
til variables can start with `$`.  What's the rhyme and reason behind when to do this?

Reply:
I'm following @yjbanov's pattern here, which if I understand it correctly is to name variables with a dollar sign if they are placeholders for literals (and to name them as close to the literal value as possible), existing only so that we can work around the constness lints.